### DISCUSSION TITLE
GraphQL Support

### DISCUSSION THREAD

**Discussion by andrewfairlie** [nmaier95/craft-shopify-product-fetcher#49] [Ideas] [answered]
There is an issue here at https://github.com/craftcms/shopify/issues/48 but I opened this thread because I don't think it's a bug, just not something currently in.

---

Is adding GraphQL support on the roadmap? The current Element API support is nice, but for sites that already use GraphQL with fragments for Matrix fields etc. writing and maintaining a separate Element API feels like a chore.

**Comment by dawb** [comment]
It would be great if this module could support GraphQL, rather than having to maintain a separate API. If it's on the roadmap, would it be possible to provide an idea of when it might be released?

**Comment by justenh** [comment]
We are performing a number of queries using a separate Storefront client (customerAccessTokens, managing customer info etc..) It would be great if the plugin provided initial support for accessing the Storefront API.

**Comment by edoardolunardi** [comment]
@lukeholder is there a roadmap for this? 🙏

**Comment by edoardolunardi** [comment]
Any update?

**Comment by plcdnl** [comment]
Looking for this feature too. Any news?

**Comment by JayBox325** [comment]
+1 interested in this.

**Comment by stephanedemotte** [comment]
@nfourtythree we need this :)

**Comment by nfourtythree** [comment]
Hi All

I believe this has been answered on other issues but it will be good to put it in this discussion.

We are aware of this being a requested feature and it is going to be something we look into after the `6.0` release of the plugin. That version is a fairly major refactor (due to Shopify's change in API requirement) and therefore it is best to look at the GraphQL part when the dust has settled on that.

We will be sure to update this discussion with any further developments and progress.

Thanks!

**Comment by lukeholder** [comment]
Although we dont have Craft element GraphQL query support, Shopify 6 now uses GraphQL to interact with shopify, so you can do `\craft\shopify\services\Api::getGqlClient()` to get the Shopify API client to make any Shopify GraphQL API request you want.

We do know the community wants Craft element GraphQL query support and we will update here once we have any news.

**Reply by plcdnl** [reply]
Hi @lukeholder, thank you.
The problem with GraphQL is that, at the moment, if you have a relationship with a product and you want to get the main informations – but also only include the ID – it doesn't work. It returns an empty string or an empty array, just like the original issue described.

https://github.com/craftcms/shopify/issues/48

**Comment by dgrigg** [comment]
Any update on this? We have products synced from Shopify and have some additional data from the Craft side. It would be nice to have that all available with GraphQL accessing the Craft Shopify Products instead of having to create a Product entry and then having to manually setup and map products that way.

**Reply by plcdnl** [reply]
Same here! For now, I solved it by creating a small module that exposes some APIs to handle this, but it would definitely be cleaner to do it directly with GraphQL.

**Comment by nfourtythree** [comment, marked as answer]
Shopify for Craft 7.1.0 has just been released which adds support for querying for Shopify Product elements in GraphQL.
